src/services/reportsService.ts

```diff
@@ -0,0 +1,19 @@
+import { Reports } from "../data/reports/report";
+import { apiSlice } from "./apiSlice";
+
+export const reportService = apiSlice.injectEndpoints({
+    endpoints: (builder) => ({
+        postReports: builder.mutation<Reports, any>({
+            query: (data) => ({
+                url: '/reports/patientsDates',
+                method: "POST",
+                body:{...data}
+            }),
+            transformResponse: (response: { data: Reports}, meta, arg) => response.data,
```

Comment:
here you already cast the type of data so in the class you don't need to cast it again

Reply:
I don't understand, with the cast again